server/src/main/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregator.java

```diff
@@ -291,6 +291,10 @@ private boolean subAggsNeedScore() {
 
     @Override
     protected boolean shouldDefer(Aggregator aggregator) {
-        return collectMode == SubAggCollectionMode.BREADTH_FIRST && !aggsUsedForSorting.contains(aggregator);
+        // don't defer when StarTreeContext is set, don't defer when collectMode == SubAggCollectionMode.BREADTH_FIRST
+        // this boolean condition can be further simplified but affects readability.
+        return (context.getQueryShardContext().getStarTreeQueryContext() == null || collectMode != SubAggCollectionMode.BREADTH_FIRST)
+            && collectMode == SubAggCollectionMode.BREADTH_FIRST
+            && !aggsUsedForSorting.contains(aggregator);
```

Comment:
Could this be:

```
if (context.getQueryShardContext().getStarTreeQueryContext() == null) {
  return false;
} else {
  return collectMode == SubAggCollectionMode.BREADTH_FIRST && !aggsUsedForSorting.contains(aggregator);
}
```

That is, make it even more complicated for readability. 😁

Reply:
Cool, refactored.